sql/structured.go

```diff
@@ -117,12 +137,29 @@ func (desc *TableDescriptor) AllocateIDs() error {
 		desc.Columns[i].ID = columnID
 	}
 
+	// Keep track of index names and unnamed indexes.
+	indexNames := make(map[string]struct{})
```

Comment:
I think this is an over optimization. Just use `TableDescriptor.FindIndexByName`. I'm not entirely sure it is correct either because you haven't passed the index names through `normalizeName`.